pkg/pillar/types/smarttypes.go

```diff
@@ -1,85 +1,132 @@
+	SmartAttrIDTemperatureCelsius int = 194
+	// SmartAttrIDRealLocatedEventCount - count of remap operations. Shows the total count of attempts.
+	SmartAttrIDRealLocatedEventCount int = 196
+)
+
+//DAttrTable have smart attr received via API
+type DAttrTable struct {
+	ID       int
+	Value    int
+	Worst    int
+	Flags    int
+	RawValue int
 }
 
-//Flags of each SMART attribute
-type Flags struct {
-	Value         int64  `json:"value"`
-	String        string `json:"string"`
-	PreFailure    bool   `json:"prefailure"`
-	UpdatedOnline bool   `json:"updated_online"`
-	Performance   bool   `json:"performance"`
-	ErrorRate     bool   `json:"error_rate"`
-	EventCount    bool   `json:"event_count"`
-	AutoKeep      bool   `json:"auto_keep"`
+//DiskSmartInfo have smart data received via API
+type DiskSmartInfo struct {
+	DiskName         string   // /dev/sda
+	DiskType         DiskType //SATA, SCSI, NVME    enum
+	ModelNumber      string   // Intel ...
+	SerialNumber     string
+	Wwn              uint64
+	SmartAttrs       []*DAttrTable // Temperature, PowerOnTime, PowerCycleCount ...
+	TimeUpdate       uint64        // Date last collect info in seconds
+	Errors           error         // errors in data collection
+	CollectingStatus CollectingStatus
 }
 
-//Raw value of each SMART attribute
-type Raw struct {
-	Value  int64  `json:"value"`
-	String string `json:"string"`
+// DisksInformation main struct for SMART
+type DisksInformation struct {
+	Disks []*DiskSmartInfo
 }
 
-//PowerOnTime of the disk
-type PowerOnTime struct {
-	Hours int64 `json:"hours"`
+// GetTemperature returns the disk temperature in degrees Celsius
+func (dsi DiskSmartInfo) GetTemperature() uint32 {
+	for _, attr := range dsi.SmartAttrs {
+		if attr.ID == SmartAttrIDTemperatureCelsius {
+			return uint32(attr.RawValue)
+		}
+	}
+	return 0
 }
 
-//Temperature of the disk
-type Temperature struct {
-	Current int64 `json:"current"`
+// GetPowerOnTime returns count of hours in power-on state
+func (dsi DiskSmartInfo) GetPowerOnTime() uint32 {
+	for _, attr := range dsi.SmartAttrs {
+		if attr.ID == SmartAttrIDPowerOnHours {
+			return uint32(attr.RawValue)
+		}
+	}
+	return 0
 }
 
-//SmartCtl query result.
-type SmartCtl struct {
-	//ExitStatus non 0 ExitStatus indicates that SMART is not available/disables
-	ExitStatus int64 `json:"exit_status"`
+// GetPowerCycleCount returns  count of full hard disk power on/off cycles.
+func (dsi DiskSmartInfo) GetPowerCycleCount() uint32 {
+	for _, attr := range dsi.SmartAttrs {
+		if attr.ID == SmartAttrIDPowerCycleCount {
+			return uint32(attr.RawValue)
+		}
+	}
+	return 0
 }
 
-//NewSmartDataWithDefaults returns 'SmartData' with default values
-func NewSmartDataWithDefaults() *SmartData {
-	return &SmartData{
-		PowerOnTime:     PowerOnTime{Hours: -1},
-		PowerCycleCount: -1,
+// GetSmartAttrViaID takes an attribute ID as input,
+// returns the properties of this attribute.
+func (dsi DiskSmartInfo) GetSmartAttrViaID(id int) *DAttrTable {
```

Comment:
This function isn't currently used. Would it make sense to express the above Get functions by using GetSmartAttrViaId()? That makes it easier to add future Get functions without having a loop in each of them, and also makes it possible to later change the implementation to have dsi.SmartAttrs be e.g., a map for faster access.